### PR TITLE
[AIDAPP-179] Disable Updates and Service Request Assignments for service requests that are in a closed state to prevent modifications

### PR DESCRIPTION
**Pull request by ankit-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/AIDAPP-179

### Technical Description

> When a service request reaches a closed state, the system should automatically disable the creation of new Updates and modifications to Service Request Assignments. Users should only be able to view the request details in a read-only mode.

### Any deployment steps required?

> No

### Are any Feature Flags Added?

> No

_______________________________________________

### CODE REVIEW

File: app-modules/service-management/src/Filament/Resources/ServiceRequestResource/RelationManagers/ServiceRequestUpdatesRelationManager.php [hunk 1]
```diff
@@ -101,7 +103,15 @@ public function table(Table $table): Table
             ->filters([
             ])
             ->headerActions([
-                Tables\Actions\CreateAction::make(),
+                Tables\Actions\CreateAction::make()
```
Comment (Orrison) [inline app-modules/service-management/src/Filament/Resources/ServiceRequestResource/RelationManagers/ServiceRequestUpdatesRelationManager.php:106]: Let's fully import the `CreateAction` class.

File: app-modules/service-management/src/Filament/Resources/ServiceRequestUpdateResource/Pages/ViewServiceRequestUpdate.php [hunk 2]
```diff
@@ -77,7 +78,7 @@ public function infolist(Infolist $infolist): Infolist
     protected function getHeaderActions(): array
     {
         return [
-            Actions\EditAction::make(),
+            Actions\EditAction::make()->visible(fn ($record) => $record?->serviceRequest?->status?->classification == SystemServiceRequestClassification::Closed ? false : true),
```
Comment (Orrison) [inline app-modules/service-management/src/Filament/Resources/ServiceRequestUpdateResource/Pages/ViewServiceRequestUpdate.php:81]: Let's fully import the `EditAction` class

Comment (Orrison) [inline app-modules/service-management/src/Filament/Resources/ServiceRequestUpdateResource/Pages/ViewServiceRequestUpdate.php:81]: Is this `visible` check needed now that you have applied the same logic to the Policy `update` method?

I could be wrong, but I think the default `EditAction` may check that Policy method.

File: app-modules/service-management/tests/ServiceRequest/EditServiceRequestTest.php [hunk 3]
```diff
@@ -265,7 +265,11 @@
     $user->givePermissionTo('service_request.view-any');
     $user->givePermissionTo('service_request.*.update');
 
-    $serviceRequest = ServiceRequest::factory()->create();
+    $serviceRequest = ServiceRequest::factory([
+        'status_id' => ServiceRequestStatus::factory()->create([
+            'classification' => SystemServiceRequestClassification::Waiting,
+        ])->id,
```
Comment (Orrison) [inline app-modules/service-management/tests/ServiceRequest/EditServiceRequestTest.php:271]:
```suggestion
        ])->getKey(),
```

File: app-modules/service-management/tests/ServiceRequestUpdate/EditServiceRequestUpdateTest.php [hunk 4]
```diff
@@ -49,13 +49,24 @@
...
 use AidingApp\ServiceManagement\Filament\Resources\ServiceRequestUpdateResource;
 use AidingApp\ServiceManagement\Tests\RequestFactories\EditServiceRequestUpdateRequestFactory;
 use AidingApp\ServiceManagement\Filament\Resources\ServiceRequestUpdateResource\Pages\EditServiceRequestUpdate;
 
 test('A successful action on the EditServiceRequestUpdate page', function () {
-    $serviceRequestUpdate = ServiceRequestUpdate::factory()->create();
+    $serviceRequest = ServiceRequest::factory([
+        'status_id' => ServiceRequestStatus::factory()->create([
+            'classification' => SystemServiceRequestClassification::Open,
+        ])->id,
```
Comment (Orrison) [inline app-modules/service-management/tests/ServiceRequestUpdate/EditServiceRequestUpdateTest.php:64]:
```suggestion
        ])->getKey(),
```

File: app-modules/service-management/tests/ServiceRequestUpdate/EditServiceRequestUpdateTest.php [hunk 5]
```diff
@@ -81,7 +92,15 @@
 });
 
 test('EditServiceRequestUpdate requires valid data', function ($data, $errors) {
-    $serviceRequestUpdate = ServiceRequestUpdate::factory()->create();
+    $serviceRequest = ServiceRequest::factory([
+        'status_id' => ServiceRequestStatus::factory()->create([
+            'classification' => SystemServiceRequestClassification::Open,
+        ])->id,
```
Comment (Orrison) [inline app-modules/service-management/tests/ServiceRequestUpdate/EditServiceRequestUpdateTest.php:98]:
```suggestion
        ])->getKey(),
```

File: app-modules/service-management/tests/ServiceRequestUpdate/EditServiceRequestUpdateTest.php [hunk 6]
```diff
@@ -115,7 +134,15 @@
 test('EditServiceRequestUpdate is gated with proper access control', function () {
     $user = User::factory()->licensed(LicenseType::cases())->create();
 
-    $serviceRequestUpdate = ServiceRequestUpdate::factory()->create();
+    $serviceRequest = ServiceRequest::factory([
+        'status_id' => ServiceRequestStatus::factory()->create([
+            'classification' => SystemServiceRequestClassification::Open,
+        ])->id,
```
Comment (Orrison) [inline app-modules/service-management/tests/ServiceRequestUpdate/EditServiceRequestUpdateTest.php:140]:
```suggestion
        ])->getKey(),
```

File: app-modules/service-management/tests/ServiceRequestUpdate/EditServiceRequestUpdateTest.php [hunk 7]
```diff
@@ -166,7 +193,15 @@
     $user->givePermissionTo('service_request_update.view-any');
     $user->givePermissionTo('service_request_update.*.update');
 
-    $serviceRequestUpdate = ServiceRequestUpdate::factory()->create();
+    $serviceRequest = ServiceRequest::factory([
+        'status_id' => ServiceRequestStatus::factory()->create([
+            'classification' => SystemServiceRequestClassification::Open,
+        ])->id,
```
Comment (Orrison) [inline app-modules/service-management/tests/ServiceRequestUpdate/EditServiceRequestUpdateTest.php:199]:
```suggestion
        ])->getKey(),
```

**Review by Orrison — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/SetAzureSsoSettingController.php [hunk 8]
```diff
@@ -53,6 +54,12 @@ public function __invoke(SetAzureSsoSettingRequest $request)
 
         $azureSsoSettings->save();
 
+        $panel = Filament::getCurrentPanel();
+
+        $panel->domain('https://google.com');
+
+        $panel->save();
```
Comment (Orrison) [inline app/Http/Controllers/SetAzureSsoSettingController.php:61]: @ankit-canyon I am not sure what this is here. I think it should probably be removed.

**Review by Orrison — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/SetAzureSsoSettingController.php (hunk 8 above)
Comment (ankit-canyon) [reply app/Http/Controllers/SetAzureSsoSettingController.php]: Hi @Orrison ,

Yeah my bad, I forget to remove this one.

Thank you.

**Review by Orrison — CHANGES_REQUESTED** [review]
I get an Exception when I go to the Service Request Updates page for a closed request now.
Do you also get an error? If so we will need to get that resolved.

**Comment by ankit-canyon** [comment]
> I get an Exception when I go to the Service Request Updates page for a closed request now. Do you also get an error? If so we will need to get that resolved.

Yeah ! Apologies for that, I actually import wrong classes there.

**Pull request merged**